src/main/java/org/openstreetmap/atlas/geography/boundary/CountryBoundaryMap.java

```diff
@@ -1347,7 +1350,34 @@ public int size()
         for (final Polygon candidate : candidates)
         {
             RuntimeCounter.geometryCheckedIntersect();
-            final Geometry clipped = target.intersection(candidate);
+            Geometry clipped;
+            try
+            {
+                clipped = target.intersection(candidate);
+            }
+            catch (final TopologyException exc)
+            {
+                logger.error(
+                        "Error while using regular intersection for line {}, attempting again with reduced precision",
+                        identifier);
```

Comment:
Here I would add the exception details too!

```suggestion
                        identifier, exc);
```